Detour/Source/DetourNavMeshQuery.cpp

```diff
@@ -721,8 +721,13 @@ dtStatus dtNavMeshQuery::findNearestPoly(const float* center, const float* exten
 	if (dtStatusFailed(queryPolygons(center, extents, filter, polys, &polyCount, MAX_SEARCH)))
 		return DT_FAILURE | DT_INVALID_PARAM;
 	
+	if (polyCount == 0)
+		return DT_SUCCESS;
```

Comment:
Slight behaviour change here; previously `nearestRef` would have been set to 0 if `queryPolygons` returned no polys. We should probably keep that behaviour.